utils/table_to_annotations.py

```diff
@@ -0,0 +1,277 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-t",
+        type=str,
+        help="Synapse Id of an dataset with files to annotate",
+        required=True
+    )
+    parser.add_argument(
+        "-f",
+        type=str,
+        help="Synapse Id of a table containing File View metadata.",
+        required=True
+    )
+    parser.add_argument(
+        "-s",
+        type=str,
+        help="Synapse Id of a table containing Biospecimen metadata.",
+        required=True
+    )
+    parser.add_argument(
+        "-i",
+        type=str,
+        help="Synapse Id of a table containing Model metadata.",
+        required=False
+    )
+    parser.add_argument(
+        "-m",
+        type=str,
+        help="Synapse Id of a table containing Model metadata.",
+        required=False
+    )
+    return parser.parse_args()
+
+
+def get_dataset_file_ids(syn, dataset_id:str) -> list:
+    """Collect file Synapse Ids from a Synapse Dataset entity"""
+    
+    query = f"SELECT id FROM {dataset_id}"
+    file_table = syn.tableQuery(query).asDataFrame().fillna("")
+    files = list(file_table["id"])
+    print(f"Synapse Ids acquired from Dataset {dataset_id}")
+
+    return files
+
+
+def get_annotations_table(syn, source_id:str) -> pd.DataFrame:
+    """Collect all entries from a Synapse Table entity"""
+    
+    query = f"SELECT * FROM {source_id}"
+    annotations_table = syn.tableQuery(query).asDataFrame().fillna("")
+    print(f"Annotations acquired from Synapse table {source_id}")
+
+    return annotations_table
+
+
+def collect_fileview_annotations(syn, files: list, fileview_id: str) -> dict:
+    """Collect all entries from a File View metadata table,
+    extract Biospecimen and File identifiers,
+    return a Biospecimen Key: File Synapse Id dictionary"""
+    
+    id_key_tuples = []
+
+    fileview_table = get_annotations_table(syn, fileview_id)
+
+    for _, row in fileview_table.iterrows():
+        id_key_tuple = (row["Biospecimen Key"], row["FileView_id"])
+        id_key_tuples.append(id_key_tuple)
+    
+    file_tuples = dict([tup for tup in id_key_tuples if tup[1] in files])
+
+    return file_tuples
+            
+
+def collect_biospecimen_annotations(syn, file_tuples: dict, specimen_info_tuple: tuple[str, str, str]) -> tuple[dict, dict]:
+    """Collect all entries from a Biospecimen metadata table,
+    select entries where Biospecimen_id is present in Biospecimen Keys associated with files,
+    match Biospecimen metadata with column names to create key:value pairs,
+    apply annotations to each file,
+    return Individual Key:File Id and Model Key: File Id dictionaries
+    """
+    
+    individual_tup_list = []
+    model_tup_list = []
+    
+    component, table_id, column_list = specimen_info_tuple
+    data_table = get_annotations_table(syn, table_id)
+    count = 0
+    for _, row in data_table.iterrows():
+        id = row["Biospecimen_id"]
+        if id in file_tuples.keys():
```

Comment:
nit: the default behavior for iterating and checking membership in a dictionary is to look at the keys, so there's no need to include `.keys()` here!  That is, this would achieve the same thing:

```suggestion
        if id in file_tuples:
```

^^ this is a bit more Pythonic as well, since it doesn't create a new list like `keys()` would.

Reply:
That makes sense!